src/providers/twitch/api/Helix.hpp

```diff
@@ -384,6 +384,59 @@ struct HelixModerators {
     }
 };
 
+struct HelixBadgeVersion
+{
+    QString id;
+    Url imageURL1x;
+    Url imageURL2x;
+    Url imageURL4x;
+    QString title;
+    Url clickURL;
+
+    explicit HelixBadgeVersion(QJsonObject jsonObject)
+        : id(jsonObject.value("id").toString())
+        , imageURL1x(Url{jsonObject.value("image_url_1x").toString()})
+        , imageURL2x(Url{jsonObject.value("image_url_2x").toString()})
+        , imageURL4x(Url{jsonObject.value("image_url_4x").toString()})
+        , title(jsonObject.value("title").toString())
+        , clickURL(Url{jsonObject.value("click_url").toString()})
+    {
+    }
+};
+
+struct HelixBadgeSet
+{
+    QString setID;
+    std::vector<HelixBadgeVersion> versions;
+
+    explicit HelixBadgeSet(const QJsonObject& json)
+        : setID(json.value("set_id").toString())
+    {
+        const auto json_versions = json.value("versions").toArray();
```

Comment:
Doesn't affect much, but we can use `json_versions.size()` to reserve space in the `versions` vector before pushing into it. Same for the HelixGlobalBadges below. 

I noticed that other constructors in this file that use vectors *don't* reserve. Maybe we should update the others (in another PR)? Would like to hear another opinion on whether to reserve.

Reply:
+1 for follow-up PR that does `reserve` across vector usage in this file

Reply:
Considering that it is not necessary for this PR, it will be better to solve it in another PR 😁👍